### ISSUE TITLE
Units are hostile or friendly

### ISSUE THREAD

**Issue by olistic** [olistic/warriorjs#61]
Units are "hostile" or "friendly", instead of "enemy" or "captive". This gives greater flexibility.

With this change:

* Unit's constructor receives a `hostile` parameter that defaults to `true` instead of the [current `captive` parameter](https://github.com/olistic/warriorjs/blob/6b90d0629e563ea664e822acf3914353c99c5a7c/packages/warriorjs-core/src/Unit.js#L18).
* [Space's `.isEnemy()`](https://github.com/olistic/warriorjs/blob/master/packages/warriorjs-core/src/Space.js#L60-L67) is removed in favor of a new and simplified method `.isHostile()`.
  * `.isHostile()` calls `unit.isHostile()` (also new) under the hood.
  * `.isHostile()` still returns `false` if the unit is bound.
* [Space's `.isCaptive()`](https://github.com/olistic/warriorjs/blob/master/packages/warriorjs-core/src/Space.js#L95-L98) is removed in favor of a new method `.isFriendly()`.
* The rescue ability should [check if the receiver is friendly, instead of captive](https://github.com/olistic/warriorjs/blob/6b90d0629e563ea664e822acf3914353c99c5a7c/packages/warriorjs-abilities/src/rescue.js#L15).

Docs and official levels tips like [this one](https://github.com/olistic/warriorjs/blob/master/packages/warriorjs-tower-beginner/src/index.js#L219) will need update as well.

**UPDATE:** Let's also pass a `bound` parameter that defaults to `false` to the Unit's constructor, and set it to `true` in the `Captive` unit definition.

**Comment by froehlichA** [comment]
If you plan to support additional friendly characters that are friendly, but not captive (for example, NPC shopkeepers) further down the road, I would suggest adding the ``.isFriendly()`` next to the ``.isCaptive()`` method, so that you have two super types: "hostile" and "friendly", and "captive" as a sub type of "friendly".

**Comment by olistic** [comment]
Hello @froehlichA! Thanks for your interest in the project. Regarding this issue, my plan is to make the core as generic as possible. For that matter, being able to distinguish between hostile and friendly units is enough, leaving the semantics of what a captive is to the towers (some towers may not even define them). For the existing towers, for example, a captive is a unit that is friendly and is bound. With that in mind, do you think there's a benefit for preserving `.isCaptive()`?

**Comment by froehlichA** [comment]
Currenty, the beginner tower uses the units imported from warriorjs/units. Why are these units defined in this seperate package, and not in the beginner tower source? Are they there for other tower creators to expand upon?

If you want to make the core as generic as possible, I would just put basic unit types, that provide a basis to build upon, in the @warriorjs/units package and leave the rest to the tower creators (like ``Archer``, ``Sludge``, or ``Wizard``).

Concerning the ``.isCaptive()`` method, my personal opinion is that the ``.isCaptive()`` should be removed from the Space API and replaced with the ``.isFriendly()`` method, like you mentioned, but that tower creators should then be able to define the appropiate methods on the units themselves instead.

An Example:
```js
//warrior-tower-community/index.js
const Shopkeeper = {
  name: 'Shopkeeper',
  character: '$',
  maxHealth: 7,
  friendly: true,
  detectionName: 'isShopkeeper'
}
```
_Warriorjs should then automatically expose the method `unit.isShopkeeper()` (or something similar) in their API._

With this principle, the tower creators themselves would be able to give players the possibility to check if an unit is a specific unit type or not, because players may want to take different actions if a unit is a friendly shopkeeper, a friendly captive or another friendly unit.

TLDR: Generally, I agree with you that there should only be detection of alignment (``.isHostile()`` and ``.isFriendly()``) in the core, but also leave detection of unit types (``.isCaptive()``, ``.isArcher()``, ``...``) to the tower creators.
